lamindb_setup/_setup_user.py

```diff
@@ -43,15 +43,35 @@ def load_user(email: str | None = None, handle: str | None = None) -> UserSettin
 
 
 def login(
-    user: str | None = None, *, api_key: str | None = None, key: str | None = None
+    user: str | None = None, *, api_key: str | None = None, **kwargs
 ) -> UserSettings:
     """Log in user.
 
     Args:
-        user: handle or email
-        api_key: API key
-        key: legacy API key
+        user: User handle.
+        api_key: User API key. The CLI `lamin login` command prompts if `api_key` is `None`.
     """
+    if user is None:
+        if api_key is None:
+            if "LAMIN_API_KEY" in os.environ:
+                api_key = os.environ["LAMIN_API_KEY"]
+            else:
+                api_key = input("Your API key: ")
+    else:
+        api_key = None
```

Comment:
I don't think this is correct, it should raise an error, otherwise confusing in my opinion, it is not immediately clear for a user if it uses `user` or `api_key` if both provided.

Reply:
overall the logic should be restructured because further checks below like `if user is None and api_key is None` don't make much sense anymore i think.

Reply:
I see that just below these sanity checks are already happening. There was some duplication across the CLI and the Python API implementations. That's why I moved the logic out of lamin-cli.

<img width="778" height="484" alt="image" src="https://github.com/user-attachments/assets/1a78aadb-0c2f-4cc1-99d7-62ab47f94e42" />

I'll clean things up.

Reply:
I have made some edits, feel free to revert if you don't agree.

Reply:
sorry, i tried to push mine but it was rejected due to your commit.

Reply:
Ohhhh, I mean i didn't spend much thought in my commits; can you take a look and add what you think is missing?

Reply:
there are still some strange things in your recent commit, can i force push my fixes?